### ISSUE TITLE
Accessibility:  Review existing heading issues and propose a plan for updating heading structure

### ISSUE THREAD

**Issue by VanessaVenti** [projectblacklight/spotlight#3535]
Review the following tickets and propose a plan to update the heading structure:

- https://github.com/projectblacklight/spotlight/issues/3525
- https://github.com/projectblacklight/spotlight/issues/3434
- https://github.com/projectblacklight/spotlight/issues/3473

**Comment by taylor-steve** [comment]
For pages:

What are folk's thoughts about the intent/current usage of the heading block?

Assume that all non-heading block (e.g., Item Row) headings are `h3`, and the heading block is an `h2`. For the sake of this discussion, assume the structural consideration with other headings outside the context of a `page` (collection of blocks) aren't an issue (which they are).

Given a page with the following blocks, where 'Block *' is any non-heading block:
* Heading 1
* Block 1
* Block 2
* Heading 2
* Block 3
* Block 4

Currently that yields a single section of content, which has invalid heading structure:
* h2 - Heading 1
* h3 - Block 1
* h3 - Block 2
* h2 - Heading 2
* h3 - Block 3
* h3 - Block 4

We could treat the heading blocks as inferred section boundaries:
* h2 - Heading 1
  * h3 - Block 1
  * h3 - Block 2
* h2 - Heading 2
  * h3 - Block 3
  * h3 - Block 4

We could treat leading heading blocks differently:
* h2 - Heading 1
  * h3 - Block 1
  * h3 - Block 2
  * h3 styled as h2 - Heading 2
  * h3 - Block 3
  * h3 - Block 4

We could keep the flat structure and differentiate heading blocks by visual style:
* h3 styled as h2 - Heading 1
* h3 - Block 1
* h3 - Block 2
* h3 styled as h2 - Heading 2
* h3 - Block 3
* h3 - Block 4

What is the intent of the heading block? Is that intent inherent to Spotlight or does it vary by implementing institution/exhibit?

**Comment by alundgard** [comment]
Here is a proposal:
- Leave the exhibit title in the masthead as h1.
- Change the page title h1 to h2. 
- Visually style the page title h2 to look more like an h1 (i.e., keep the slightly larger font size).
- If a user has unchecked the "Show title" box for that page, visually-hide the title but still render the title h2 in the DOM.
- If a page doesn't have any title, visually-hide but render a generic "Main content" title h2 in the DOM.
- Leave the within-page headings (h2) and widget headings (h3) as they are.
